include/oneapi/dpl/internal/dynamic_selection_impl/auto_tune_policy.h

```diff
@@ -229,6 +229,9 @@ class auto_tune_policy
             }
             else
             {
+                if constexpr(lazy_report_v<Backend>){
+                        backend_->lazy_report();
```

Comment:
You are holding a lock on state_->m_ when calling the lazy_report function. Seems unnecessary and perhaps dangerous as a general pattern.

Reply:
Fixed